### PR TITLE
Integrate feedback export functionality with the feedback view filters

### LINKED ISSUE #27426: Integrate feedback export button into with the feedback table filters
In order to provide a more streamlined and intuitive experience, we should remove the 'download feedback form' from underneath the feedback table in WP Admin and replace it with a single 'Export' button at the top of the table which would export the current view or selection.

### PR DESCRIPTION
**Pull request by ice9js**
Fixes #27426.

#### Changes proposed in this Pull Request:
This PR integrates the export/download feedback functionality with the WP admin table by the following:

- A new second 'post' dropdown has been added at the top of the table in the filter section, which allows filtering feedback by the post from which it originates, in the same way as you'd select a post to download feedback from.
- An export button has been added at the top of the table, which by default will export all results matching the currently active filter. If any rows are selected, only those rows will be exported.
- The download feedback form at the bottom of the page has been removed as it's now obsolete.

Old export form:
Screenshot 2022-11-15 at 22 13 20

New, integrated export form:
Screenshot 2022-11-15 at 22 12 54

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

pcsBup-oK-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

- Go to WP Admin -> Feedback.
- You should now be able to filter your feedback responses by the post by choosing from the dropdown menu at the top of the table.
- Clicking 'Export' at the top of the table should export all responses matching the currently applied filter.
- Select some rows.
- When a selection is active, only the selected rows should be exported.

### CODE REVIEW

**Comment by digitalwaveride** [comment]
Thanks so much Kuba!!!

Unfortunately, I didn't see the Export button. This is was got via Jurassic Ninja:

image



And one question: 
Is there any chance that we can customize the filter more specific to the "forms/source"?

In this version, the filter is focused on "posts" (... which actually also includes "pages" 🤷 , anyways... ).

This comes with a few issues:

- In the dropdown, it shows the title of the posts/pages. If the title of a page is empty, the dropdown is empty.

image


- More importantly, we don't show in this responses table the title at all, we show instead a source column. So users have no reference to what they are actually filtering for.

image


Any chance we can rename that filter to "All sources" - and in the dropdown showing the individual sources?

File: projects/plugins/jetpack/modules/contact-form/admin.php
Comment (jcheringer) [inline projects/plugins/jetpack/modules/contact-form/admin.php]: Because of the validation above:

```
	if ( ! defined( 'AKISMET_VERSION' ) ) {
		return;
	}
```

The Export button is only visible if the Akismet plugin is active.
Was that intentional?

Comment (ice9js) [reply projects/plugins/jetpack/modules/contact-form/admin.php]: Nice catch, fixed 👍

**Comment by ice9js** [comment]
@digitalwaveride I updated the source filter dropdown to say `All Sources` instead of `All Posts` as well as provide a default `(Untitled)` text for options where the post title is empty.

**Comment by digitalwaveride** [comment]
Thanks a ton Kuba!
Export worked as advertised! 👍 

Regarding the "All sources" dropdown, nice 🙏, on more thing there: Please also update inside of the dropdown the "Source" values.

**Review by jcheringer — DISMISSED** [review]
LGTM

**Comment by ice9js** [comment]
I updated the sources dropdown to match the upcoming 'source' column values from #27368.

**Review by digitalwaveride — DISMISSED** [review]
Awesome!
Things worked for me as expected!

Kuba, thanks a ton!!!

**Review by CGastrell — DISMISSED** [review]
Worked well on simple site, AT and self hosted :shipit:

File: projects/plugins/jetpack/modules/contact-form/admin.php
Comment (coder-karen) [inline projects/plugins/jetpack/modules/contact-form/admin.php:485]: What is the reason why the nonce sniff is being ignored here? It looks like a returned GET value is being used in `form_posts_dropdown`, so it looks like something is being changed in response to the request? (referencing this post: p9dueE-4z8-p2 )

File: projects/plugins/jetpack/modules/contact-form/admin.php
Comment (coder-karen) [inline projects/plugins/jetpack/modules/contact-form/admin.php:498]: Same as the other review, in this case the value may be added to `$query`.

**Review by coder-karen — COMMENTED** [review]
Tests well! I left some reviews, mainly referencing this: p9dueE-4z8-p2

File: projects/plugins/jetpack/modules/contact-form/admin.php
Comment (ice9js) [reply projects/plugins/jetpack/modules/contact-form/admin.php:485]: It's a `GET` request and so it's not supposed to be changing anything, the value is being used just to make sure the dropdown's default options matches the currently applied filter. So I guess it falls under 'not taking any action' - should I add a relevant comment in there?

File: projects/plugins/jetpack/modules/contact-form/admin.php
Comment (ice9js) [reply projects/plugins/jetpack/modules/contact-form/admin.php:498]: I could be wrong here, but again, it's just supposed to be a filter for the results in the view - it's not creating/updating anything.

**Comment by ice9js** [comment]
I hope my comments add some clarity, but let me know if I'm still overlooking something why the nonce should be there.  

To give a counterargument, to me, the whole idea of using query params here is that you can copy the URL, share it with someone, and they'll get exactly the same view (assuming they also have access to the feedback area).  
I don't necessarily see any risks with that but then again, I can only speak for my code and I might not be fully aware of the larger implications for Jetpack and/or WP.

File: projects/plugins/jetpack/modules/contact-form/admin.php
Comment (coder-karen) [reply projects/plugins/jetpack/modules/contact-form/admin.php:485]: Thanks for explaining! Yes the additional comment is a good idea here (like on line 890 of admin.php) is a good idea.

File: projects/plugins/jetpack/modules/contact-form/admin.php
Comment (coder-karen) [reply projects/plugins/jetpack/modules/contact-form/admin.php:498]: Thanks for clarifying.

**Review by coder-karen — DISMISSED** [review]
>  but let me know if I'm still overlooking something why the nonce should be there.

In this case it sounds like they aren't needed (no changes to the site in response to the field being set). I'm still learning with nonce verification but find this page - p9dueE-4z8-p2 -  super valuable (and as a result dig in more when there aren't comments explaining why a sniff is ignored, if I catch that!). 

So as to not hold up your PR any more I've rebased, fixed the linting error (some blank lines in admin.php) and everything should be good to go (assuming adding comments would be in a follow-up).

**Comment by ice9js** [comment]
I guess I might need another approval after resolving the last conflict, sorry!

**Review by coder-karen — APPROVED** [review]

**Pull request merged**
